pkg/prometheus/prometheus.go

```diff
@@ -0,0 +1,143 @@
+package prometheus
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/openshift-scale/perf-analyzer/pkg/config"
+	"github.com/openshift-scale/perf-analyzer/pkg/result"
+	"github.com/openshift-scale/perf-analyzer/pkg/stats"
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+var InsecureRoundTripper http.RoundTripper = &http.Transport{
+	Proxy: http.ProxyFromEnvironment,
+	DialContext: (&net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}).DialContext,
+	TLSClientConfig:     &tls.Config{InsecureSkipVerify: true},
+	TLSHandshakeTimeout: 10 * time.Second,
+}
+
+type TimePoint [2]float64
+type TimeSeriesPoints []TimePoint
+
+type TimeSeries struct {
+	Name   string
+	Points TimeSeriesPoints
+	Tags   []rawData
+}
+
+type rawData struct {
+	Name   string
+	Values []float64
+}
+
+type prometheusConfig struct {
+	config api.Config
+	API    v1.API
+	Errors []error
+}
+
+func newPrometheusConfig(url, token string, insecureTLS bool) prometheusConfig {
+	return prometheusConfig{config: api.Config{Address: url, Authorization: token, RoundTripper: InsecureRoundTripper}}
+}
+
+func (c *prometheusConfig) newPrometheusAPI() {
+	client, err := api.NewClient(c.config)
+	if err != nil {
+		fmt.Printf("Error creating Prometheus client: %v\n", err)
+		c.Errors = append(c.Errors, err)
+	} else {
+		c.API = v1.NewAPI(client)
+	}
+}
+
+// DoPrometheusQuery will run queries against Prometheus endpoint
+func DoPrometheusQuery(cfg config.ScrapeConfig) {
+	//cfg.UrlFlag, cfg.TokenFlag, cfg.StepFlag, cfg.DurationFlag, cfg.InsecureTLSFlag
+	// url, token, stepFlag string, durationFlag int, insecureTLS bool
+	config := newPrometheusConfig(cfg.UrlFlag, cfg.TokenFlag, cfg.InsecureTLSFlag)
+	config.newPrometheusAPI()
+	if len(config.Errors) != 0 {
+		fmt.Printf("Client Error %v\n", config.Errors[0])
+	}
+
+	queries := map[string]string{
+		"cpu":    "sum(rate(container_cpu_usage_seconds_total{job=\"kubelet\", image!=\"\", container!=\"POD\"}[5m])) by (namespace)",
+		"memory": "sum(container_memory_usage_bytes{container_name!=\"\"}) by (namespace)",
+	}
+	end := time.Now()
+	start := end.Add(time.Duration(-1*cfg.DurationFlag) * time.Minute)
+	step, err := time.ParseDuration("1m")
+	if err != nil {
+		fmt.Printf("Error parsing step duration %s\n", cfg.StepFlag)
+	}
+	r := v1.Range{Start: start, End: end, Step: step}
+
+	var results []result.ResultType
+	for resource, query := range queries {
+		queryResult, _, err := config.API.QueryRange(context.Background(), query, r)
+		if err != nil {
+			fmt.Printf("Query Error: %v\n", err)
```

Comment:
You may want to say "Prometheus query error" or something like that.  "Query Error" seems too generic to me.  Similarly below.  Any reason for uppercasing Error?

Reply:
Fixed.